teps/0152-tekton-conformance-verification.md

```diff
@@ -0,0 +1,219 @@
+
+- The black box testing of the vendor focuses solely on the outputs of the service, disregarding its internal implementation details. The vendor service takes ownership of monitoring CRD reconciliation and delivering compatible Tekton YAML outputs.
+
+- For verifying results, upon receiving [Tekton client](https://github.com/tektoncd/pipeline/blob/main/test/clients.go#L56-L71) readable YAML outputs,  the test suite will leverage the parse pkg for validating whether all the required conformant API fields are populated and the behaviors are honored.
+
+#### Vendor integration
+
+Tekton vendors will have the responsibility of integrating the conformance verification test suite with their respective services. To deliver conformance validation results, vendors must implement the `ProcessAndSendToTekton` function. On a higher level, it is a wrapper of all the vendor service detail, intaking and outputing Tekton YAML for Tekton conformance fields to get verified.
+
+This function transforms vanilla Tekton Pipeline YAMLs into Tekton-parsable YAMLs for results verification. For more details:
+
+```go
+// ProcessAndSendToTekton accepts vanilla Tekton PipelineRun and TaskRun YAMLs, waits for the object to complete and outputs the completed PipelineRun and TaskRun along with the status.
+
+// The function parameters are the inputYAML and its corresponding Primitive type (either {PipelineRun} or {TaskRun}). Additionally, the function allows for the inclusion of custom inputs as variadic parameters.
+
+// The function will return the completed output YAML string and any associated errors.
+func ProcessAndSendToTekton(inputYAML, type string, customInputs …interface{}) (string, error)
+```
+
+#### **Alternatives Considered**
+As an alternative, Tekton vendors could expose endpoints for Tekton to call, allowing for  consumption and output of results according to HTTP API conformance (out-of-scope for the existing Tekton Conformance policy). In this scenario, Tekton Pipeline could trigger the testing process autonomously.
+
+##### Cons:
+- This would require HTTP API conformance which is not currently in-scope with [v1 Tekton conformance API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md).
+- This increases complexity and maintenance costs for the Tekton OSS project. Vendors seeking conformance should be responsible for managing these costs.
+
+### Vendor Requirements
+- Vendor service should handle conformant vanilla PipelineRun and TaskRun YAMLs successfully, ensuring all the REQUIRED fields are accepted.
+- Vendor service should be able to generate PipelineRun and TaskRun output YAMLs that are also conformant. This ensures that all `REQUIRED` fields are populated, demonstrating the expected functionality.
```

Comment:
I think REQUIRED fields being populated is necessary but not sufficient. We also need to verify the status fields are actually what is expected and also some of the behavior e.g. if two tasks use the same workspace, and task A writes a file to that workspace, then task B can actually access that file

Reply:
That makes sense to be called out. Updated.